MapboxNavigation/RouteMapViewController.swift

```diff
@@ -395,124 +320,148 @@ class RouteMapViewController: UIViewController {
         }
     }
 
-    var contentInsets: UIEdgeInsets {
-        let top = self.navigationView.instructionsBannerContentView.bounds.height
-        let bottom = self.navigationView.bottomBannerView.bounds.height
-        return UIEdgeInsets(top: top, left: 0, bottom: bottom, right: 0)
-    }
-
-    // MARK: End Of Route
-
-    func embedEndOfRoute() {
-        let endOfRoute = self.endOfRouteViewController
-        addChild(endOfRoute)
-        self.navigationView.endOfRouteView = endOfRoute.view
-        self.navigationView.constrainEndOfRoute()
-        endOfRoute.didMove(toParent: self)
+    /**
+     Updates the current road name label to reflect the road on which the user is currently traveling.
 
-        endOfRoute.dismissHandler = { [weak self] _, _ in
-            self?.routeController.endNavigation()
-            self?.delegate?.mapViewControllerDidDismiss(self!, byCanceling: false)
+     - parameter location: The user’s current location.
+     */
+    func labelCurrentRoad(at rawLocation: CLLocation, for snappedLoction: CLLocation? = nil) {
+        guard self.navigationView.resumeButton.isHidden else {
+            return
         }
-    }
-
-    func unembedEndOfRoute() {
-        let endOfRoute = self.endOfRouteViewController
-        endOfRoute.willMove(toParent: nil)
-        endOfRoute.removeFromParent()
-    }
 
-    func showEndOfRoute(duration: TimeInterval = 1.0, completion: ((Bool) -> Void)? = nil) {
-        self.embedEndOfRoute()
-        self.endOfRouteViewController.destination = self.destination
-        self.navigationView.endOfRouteView?.isHidden = false
+        let roadName = self.delegate?.mapViewController(self, roadNameAt: rawLocation)
+        guard roadName == nil else {
+            if let roadName {
+                self.navigationView.wayNameView.text = roadName
+                self.navigationView.wayNameView.isHidden = roadName.isEmpty
+            }
+            return
+        }
 
-        view.layoutIfNeeded() // flush layout queue
-        NSLayoutConstraint.deactivate(self.navigationView.bannerShowConstraints)
-        NSLayoutConstraint.activate(self.navigationView.bannerHideConstraints)
-        self.navigationView.endOfRouteHideConstraint?.isActive = false
-        self.navigationView.endOfRouteShowConstraint?.isActive = true
+        let location = snappedLoction ?? rawLocation
 
-        self.mapView.enableFrameByFrameCourseViewTracking(for: duration)
-        self.mapView.setNeedsUpdateConstraints()
+        self.labelCurrentRoadFeature(at: location)
 
-        let animate = {
-            self.view.layoutIfNeeded()
-            self.navigationView.floatingStackView.alpha = 0.0
+        if let labelRoadNameCompletionHandler {
+            labelRoadNameCompletionHandler(true)
         }
+    }
 
-        let noAnimation = { animate(); completion?(true) }
+    func labelCurrentRoadFeature(at location: CLLocation) {
```

Comment:
Did anything change here or was this simply moved?

Reply:
just moved